packages/fetch-core/src/withFetchRedirectionOn403.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { useHistory, ReactOidcHistory } from '@axa-fr/react-oidc-core';
+
+export const fetchWithRedirectionOn403 = (fetch: typeof window.fetch, history: ReactOidcHistory) => async (
+  url: RequestInfo,
+  options: RequestInit = { method: 'GET' }
+) => {
+  const response = await fetch(url, options);
+  if (response.status === 403) {
+    history.push('/authentication/not-authorized');
+  }
+  return response;
+};
+
+export const wrapAuthenticating = (fetchWithRedirectioInjected: typeof fetchWithRedirectionOn403) => (fetch: typeof window.fetch) => (
+  WrappedComponent: React.ComponentType
+) => (props: any) => {
+  const history = useHistory();
+  const previousFetch = fetch || props.fetch;
```

Comment:
**question:** I don't understand what is happening here, the `fetch` you give in param is mandatory and should be a `window.fetch`, why would you need the `||`?

Reply:
I can't be more agree with you about lisibility, the first fetch is just for the tests.

I think I will work on better code lisibility later.